csharp/autogen/src/Session.cs

```diff
@@ -111,13 +107,17 @@ public Session(string url, string opaqueRef)
         /// <param name="session"></param>
         /// <param name="timeout"></param>
         public Session(Session session, int timeout)
-            : this(session.Url, timeout)
+            : this(timeout, session.Url)
         {
             opaque_ref = session.opaque_ref;
             APIVersion = session.APIVersion;
+            if (XmlRpcToJsonRpcInvoker != null)

```

Comment:
I assume there is a reason why this can't call the SetupSessionDetails the same as the other constructors?

Reply:
SetupSessionDetails makes extra requests to the server to populate the Session's properties, but this is not needed here because they can be copied from the Session object we pass in as a parameter.
Having said this, it would probably be more correct to also copy over the XmlRpcProxy and the JsonRpcClient properties instead of creating new ones, but at this time I'd rather avoid greater scale changes.